### PR TITLE
Distributive algebras (+ a bit)

### PR DESCRIPTION
**Pull request by robrix**
This PR builds on #4 to update for `ghc` 8.10, `haskeline` 0.8, & the latest `fused-effects`. Unfortunately I can’t push to that branch because _reasons_, so, separate PR.

### CODE REVIEW

**Comment by robrix** [comment]
@jkachmar: I’ve sent you an invite to join the repo as a collaborator; I’d be perfectly content to defer to your PR so you get the contrib graph bonus points before merging, but in the meantime I needed this branch up to test `semantic` against.

**Review by patrickt — APPROVED** [review]

**Pull request merged**
